### PR TITLE
added keychain to get inline with msal_flutter

### PR DESCRIPTION
**Pull request by omaralmgerbie**

### CODE REVIEW

File: lib/src/authentication_service.dart [hunk 1]
```diff
@@ -105,14 +108,16 @@ class AuthenticationService {
       _updateStatus(AuthenticationStatus.authenticating);
       await pca!.acquireToken(defaultScopes);
       _updateStatus(AuthenticationStatus.authenticated);
-    } catch (e) {
+    } on MsalException catch (e) {
+      print(e.errorMessage);
```
Comment (mswehli) [inline lib/src/authentication_service.dart:112]: Remove print statements

File: lib/src/authentication_service.dart [hunk 2]
```diff
@@ -105,14 +108,16 @@ class AuthenticationService {
...
       _updateStatus(AuthenticationStatus.failed);
+      rethrow;
     }
   }
 
-  Future logout() async {
+  Future logout([bool browserLogout = false]) async {
```
Comment (mswehli) [inline lib/src/authentication_service.dart:118]: was this on purpose, or should it have used {} intead of [] ?

File: pubspec.yaml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 name: msal_guard
 description: Guard widget for the Msal Flutter library
-version: 0.0.1-alpha.15
+version: 0.0.1-beta.1
 homepage: https://github.com/muljin/msal-guard
 repository: https://github.com/muljin/msal-guard
-
+publish_to: none
```
Comment (mswehli) [inline pubspec.yaml:6]: think this was left by mistake.

**Review by mswehli — CHANGES_REQUESTED** [review]

File: pubspec.yaml (hunk 3 above)
Comment (omaralmgerbie) [reply pubspec.yaml:6]: yeah forgot about this.

File: lib/src/authentication_service.dart [hunk 4]
```diff
@@ -105,14 +108,15 @@ class AuthenticationService {
       _updateStatus(AuthenticationStatus.authenticating);
       await pca!.acquireToken(defaultScopes);
       _updateStatus(AuthenticationStatus.authenticated);
-    } catch (e) {
+    } on MsalException catch (e) {
```
Comment (mswehli) [inline lib/src/authentication_service.dart:111]: keep this catching any exception. Regardless of the exception type it should have failed.

**Review by mswehli — APPROVED** [review]

**Comment by omaralmgerbie** [comment]
merged

**Pull request merged**